plugin/storage/es/esCleaner.py

```diff
@@ -60,13 +64,36 @@ def main():
 def filter_main_indices(ilo, prefix):
     ilo.filter_by_regex(kind='prefix', value=prefix + "jaeger")
     empty_list(ilo, "No indices to delete")
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-read'], exclude=True)
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-write'], exclude=True)
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-service-read'], exclude=True)
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-service-write'], exclude=True)
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-archive-read'], exclude=True)
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-archive-write'], exclude=True)
+    empty_list(ilo, "No indices to delete")
     # This excludes archive index as we use source='name'
     # source `creation_date` would include archive index
     ilo.filter_by_age(source='name', direction='older', timestring='%Y-%m-%d', unit='days', unit_count=int(sys.argv[1]))
 
 
+def filter_main_indices_rollover(ilo, prefix):
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-*-read'])
+    empty_list(ilo, "No indices to delete")
+    # do not remove active write indices
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-span-write'], exclude=True)
+    empty_list(ilo, "No indices to delete")
+    ilo.filter_by_alias(aliases=[prefix + 'jaeger-service-write'], exclude=True)
+    empty_list(ilo, "No indices to delete")
+    ilo.filter_by_regex(kind='prefix', value=prefix + 'jaeger-span-archive*', exclude=True)
+    empty_list(ilo, "No indices to delete")
+    # This excludes archive index as we use source='name'
+    # source `creation_date` would include archive index
+    # TODO it might be useful to allow filter_by_space
+    ilo.filter_by_age(source='creation_date', direction='older', unit='days', unit_count=int(sys.argv[1]))
```

Comment:
Just wondering about a particular scenario - if rollover criteria is based on index size (for example) in low activity environment, it might be possible that the same index is in use for more days that the `unit_count`, in which case it would remove the index - even though it is in current use?

Reply:
I think you are right removing by size does not really make sense.